Document/0x06h-Testing-Platform-Interaction.md

```diff
@@ -169,19 +169,35 @@ In order to address these attack vectors, the outcome of the following checks sh
 
 #### Overview
 
--- TODO [Provide a general description of the issue "Testing WebView Protocol Handlers".] --
+Quoting a [source](https://developer.apple.com/library/content/documentation/iPhone/Conceptual/iPhoneOSProgrammingGuide/Inter-AppCommunication/Inter-AppCommunication.html) "Apple provides built-in support for the `http`, `mailto`, `tel`, and `sms` URL schemes among others". The handlers cannot be modified or disabled. So if any of the schemes will be invoked in a WebView object the Apple-provided app is launched instead of your app.
+
+WebViews can load content remotely, but can also load it locally from the app data directory. If the content is loaded locally it should not be possible by the user to influence the filename or path where the file is loaded from or should be able to edit the loaded file.
 
 #### Static Analysis
 
--- TODO [Describe how to assess this given either the source code or installer package (APK/IPA/etc.), but without running the app. Tailor this to the general situation (e.g., in some situations, having the decompiled classes is just as good as having the original source, in others it might make a bigger difference). If required, include a subsection about how to test with or without the original sources.] --
+Check the source code for the usage of WebViews. If a WebView instance can be identified check if any local files are loaded ("example_file.html" in the below example).
 
--- TODO [Confirm purpose of remark "Use the &lt;sup&gt; tag to reference external sources, e.g. [Meyer's recipe for tomato soup](http://www.finecooking.com/recipes/meyers-classic-tomato-soup.aspx "Meyer's Recipe for Tomato Soup")."] --
+```sh
+- (void)viewDidLoad
+{
+    [super viewDidLoad];
+    WKWebViewConfiguration *configuration = [[WKWebViewConfiguration alloc] init];
+        
+    self.webView = [[WKWebView alloc] initWithFrame:CGRectMake(10, 20, CGRectGetWidth([UIScreen mainScreen].bounds) - 20, CGRectGetHeight([UIScreen mainScreen].bounds) - 84) configuration:configuration];
+    self.webView.navigationDelegate = self;
+    [self.view addSubview:self.webView];
+    
+    NSString *filePath = [[NSBundle mainBundle] pathForResource:@"test_local" ofType:@"html"];
+    NSString *html = [NSString stringWithContentsOfFile:filePath encoding:NSUTF8StringEncoding error:nil];
+    [self.webView loadHTMLString:html baseURL:[NSBundle mainBundle].resourceURL];
+}
+```
 
--- TODO [Add content on static analysis of "Testing WebView Protocol Handlers" with source code) --
+The `baseURL` should be checked, if any dynamic parameters are used that can be manipulated, which may lead to local file inclusion.
 
 #### Dynamic Analysis
 
--- TODO [Describe how to test for this issue "Testing WebView Protocol Handlers" by running and interacting with the app. This can include everything from simply monitoring network traffic or aspects of the app’s behavior to code injection, debugging, instrumentation, etc.] --
```

Comment:
There is a needle module available to test URI, never tested it though:
https://github.com/mwrlabs/needle/blob/0af937f0a0d2c8818eb9210d801f748c80e14188/needle/modules/dynamic/ipc/open_uri.py

But I am not sure if this module makes much sense, as you were saying that anyway you cannot disable for example tel://.